llvm/lib/Analysis/ValueTracking.cpp

```diff
@@ -5004,6 +5004,15 @@ void computeKnownFPClass(const Value *V, const APInt &DemandedElts,
 
       break;
     }
+      // reduce min/max will choose an element from one of the vector elements,
+      // so we can infer and class information that is common to all elements.
+    case Intrinsic::vector_reduce_fmax:
+    case Intrinsic::vector_reduce_fmin:
+    case Intrinsic::vector_reduce_fmaximum:
+    case Intrinsic::vector_reduce_fminimum:
+      computeKnownFPClass(II->getArgOperand(0), Known, InterestedClasses,
```

Comment:
You cannot keep the signbit information if we don't know it is never nan.

Reply:
I need to add proofs for all of the cases. I will do so before proceeding w/ this.

Reply:
For the fmin/fmax case you only need to know one element is non-nan

Reply:
Plan now isn't to do per-ele, although I can leave that as a todo.

Also looks like alive2 doesn't support fp reduce intrins.

Reply:
Fixed + tests.